test/provider/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,159 @@
+// +build e2e_provider
+
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+)
+
+const (
+	providerName           = "provider-gcp"
+	initialProviderPackage = "crossplane/provider-gcp:v0.16.0"
+	upgradeProviderPackage = "crossplane/provider-gcp:master"
+)
```

Comment:
@rahulgrover99 we want this to be generalized, either by passing package upgrade skews (i.e. `crospslane/provider-gcp:v0.16.0` -> `crospslane/provider-gcp:master`, etc.) in `-args` or environment variables in the `provider-upgrade` `go test` step in the GH action. Or by maintaining a config file in this repo that this test reads from. I think the latter may be a better option for maintainability and allowing provider devs who want their upgrade tested a simpler PR workflow.